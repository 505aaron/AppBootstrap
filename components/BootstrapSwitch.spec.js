import React from "react";
import { View, Text } from "react-native";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import Bootstrapper from "./BootstrapSwitch";

const Loading = () => (
  <View>
    <Text>Loading Screen</Text>
  </View>
);
const Bootstrapped = () => (
  <View>
    <Text>Loaded!</Text>
  </View>
);

describe("BootstrapSwitch", () => {
  let shallowRenderer;
  let baseProps;

  beforeEach(() => {
    baseProps = {
      initializeApp: jest.fn(),
      loading: <Loading />
    };

    shallowRenderer = new ShallowRenderer();
  });

  test("render - loading", () => {
    shallowRenderer.render(
      <Bootstrapper {...baseProps}>
        <Bootstrapped />
      </Bootstrapper>
    );

    const tree = shallowRenderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  test("render - bootstrapped", () => {
    shallowRenderer.render(
      <Bootstrapper {...baseProps} boostrapComplete>
        <Bootstrapped />
      </Bootstrapper>
    );

    const tree = shallowRenderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  test("lifecyle - mount", () => {
    const testRenderer = renderer.create(
      <Bootstrapper {...baseProps} boostrapComplete>
        <Bootstrapped />
      </Bootstrapper>
    );

    expect(baseProps.initializeApp).toHaveBeenCalledTimes(1);
  });
});
