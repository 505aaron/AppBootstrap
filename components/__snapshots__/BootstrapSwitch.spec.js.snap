// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BootstrapSwitch render - bootstrapped 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Loading />
</View>
`;

exports[`BootstrapSwitch render - loading 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Loading />
</View>
`;
